diff --git a/rc.boot b/rc.boot
index a86a29f..073f67c 100755
--- a/rc.boot
+++ b/rc.boot
@@ -2,7 +2,7 @@
 
 . PREFIX/lib/init/rc.conf
 
-welcome() { printf '\033[1;36m=>\033[m\033[1m Welcome to\033[35m Carbs Linux\033[m\033[1m!\n' ;}
+welcome() { printf '\033[1;36m=>\033[m\033[1m Welcome to\033[35m Gentoo Linux\033[m\033[1m!\n' ;}
 
 mnt() {
     mountpoint -q "$1" || {
@@ -21,7 +21,7 @@ emergency_shell() {
 }
 
 main() {
-    PATH=/usr/bin:/usr/sbin
+    PATH=/bin:/sbin:/usr/bin:/usr/sbin
     old_ifs=$IFS
     set -f
 
@@ -142,12 +142,12 @@ main() {
     }
 
     log "Setting hostname..."; {
-        read -r hostname < /etc/hostname
-        printf '%s\n' "${hostname:-carbs-linux}" > /proc/sys/kernel/hostname
+        [ -r "/etc/conf.d/hostname" ] && . /etc/conf.d/hostname
+        printf '%s\n' "${hostname:-gentoo-linux}" > /proc/sys/kernel/hostname
     } 2>/dev/null
 
     log "Getting keymap settings..."; {
-	[ -n "$keymap" ] && loadkmap < "$keymap"
+	[ -n "$keymap" ] && loadkeys "$keymap"
     }
 
     log "Loading sysctl settings..."; {
diff --git a/rc.conf b/rc.conf
index 37a414f..ca02355 100644
--- a/rc.conf
+++ b/rc.conf
@@ -5,7 +5,7 @@
 log() { printf "\033[1;36m->\033[m\033[1m %s\n\033[0;00m" "$@" ;}
 error() { printf "\033[1;31m-> ERROR: %s\n\033[0;00m" "$@" ;}
 
-# Set your keymap (full path to your keymap e.g. /usr/share/bkeymaps)
+# Set your keymap
 keymap=
 
 # Set dmesg level
